packages/compass-workspaces/src/components/workspaces.tsx

```diff
@@ -105,159 +101,119 @@ const CompassWorkspaces: React.FunctionComponent<CompassWorkspacesProps> = ({
-            tooltip.push(['Derived from', collection]);
-          } else {
-            tooltip.push(['Collection', collection]);
-          }
-          return {
-            id: tab.id,
-            connectionName,
-            type: tab.type,
-            title: collection,
-            tooltip,
-            iconGlyph:
-              collectionType === 'view'
-                ? 'Visibility'
-                : collectionType === 'timeseries'
-                ? 'TimeSeries'
-                : isNonExistent
-                ? 'EmptyFolder'
-                : 'Folder',
-            'data-namespace': ns,
-            tabTheme: getThemeOf(tab.connectionId),
-            ...(isNonExistent && {
-              className: nonExistantStyles,
-            }),
-          } as const;
+          const { isNonExistent: collectionDoesNotExist } =
+            collectionInfo[namespaceId] ?? {};
+          isNonExistent = collectionDoesNotExist;
         }
       }
-    });
-  }, [tabs, collectionInfo, databaseInfo, getThemeOf, getConnectionById]);
 
-  const activeTabIndex = tabs.findIndex((tab) => tab === activeTab);
-  const WorkspaceComponent = getWorkspacePluginByName(activeTab?.type);
+      return {
+        id: tab.id,
+        renderTab: (workspaceTabCoreProps: WorkspaceTabCoreProps) => (
+          <ErrorBoundary
+            displayName={tab.type}
+            onError={(error, errorInfo) => {
+              log.error(
+                mongoLogId(1_001_000_360),
+                'Workspace',
+                'Rendering workspace tab header failed',
+                { name: tab.type, error: error.message, errorInfo }
+              );
+            }}
+          >
+            <WorkspaceTabContextProvider tab={tab} sectionType="tab-title">
+              <Provider>
+                <WorkspaceTabTitle
+                  workspaceProps={{
+                    ...(isNonExistent !== undefined ? { isNonExistent } : {}),
+                    ...tab,
+                  }}
+                  tabProps={workspaceTabCoreProps}
+                />
+              </Provider>
+            </WorkspaceTabContextProvider>
+          </ErrorBoundary>
+        ),
+        content: (
+          <ErrorBoundary
+            displayName={tab.type}
+            onError={(error, errorInfo) => {
+              log.error(
+                mongoLogId(1_001_000_277),
+                'Workspace',
+                'Rendering workspace tab content failed',
+                { name: tab.type, error: error.message, errorInfo }
+              );
+            }}
+          >
+            <WorkspaceTabContextProvider
+              tab={tab}
+              sectionType="tab-content"
+              onNamespaceNotFound={onNamespaceNotFound}
+            >
+              <Provider>
+                <WorkspaceTabContent
+                  // Cast to any as it can be any of the workspace types
+                  // here.
+                  {...(tab as any)}
```

Comment:
If you look at the logic inside the `WorkspaceTabContextProvider` it's actually not all the props that need to be passed down, like you can do it, but as always it's better to be a bit more explicit. If you move the `Provider` inside `WorkspaceTabContextProvider` though this should be handled for you consistently